app/web/features/profile/view/leaveReference/ReferenceForm.tsx

```diff
@@ -85,7 +86,14 @@ export default function ReferenceForm({
     }));
   };
 
-  return step === "appropriate" ? (
+  const isSkippedStep =
+    referenceData.wasAppropriate === "" && step !== "appropriate";
+
+  return isSkippedStep ? (
+    <Redirect
+      to={`${leaveReferenceBaseRoute}/${referenceType}/${userId}/${hostRequestId}`}
```

Comment:
This needs an additional check on whether `hostRequestId` is defined is not for the final redirect URL. Otherwise, for friend references, I got `undefined` at the end of the URL (i.e. something like `/leave-reference/friend/4/undefined`)

Might be worth an additional test to check for this case too once fixed